src/main/java/io/passport/server/controller/PopulationController.java

```diff
@@ -75,87 +81,136 @@ public ResponseEntity<?> getPopulationByStudyId(@RequestParam Long studyId,
+
+            // Audit log
+            if (savedPopulation.getPopulationId() != null) {
+                String recordId = savedPopulation.getPopulationId().toString();
+                String description = "Creation of Population " + recordId;
+                auditLogBookService.createAuditLog(
+                        principal.getSubject(),
+                        principal.getClaim("preferred_username"),
+                        studyId,
+                        "CREATE",
+                        "Population",
+                        recordId,
+                        savedPopulation,
+                        description
+                );
+            }
             return ResponseEntity.status(HttpStatus.CREATED).body(savedPopulation);
+
         } catch (Exception e) {
-            log.error(e.getMessage());
+            log.error(e.getMessage(), e);
             return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.getMessage());
         }
     }
 
     /**
      * Update Population.
-     * @param populationId ID of the population to be updated.
+     * (Only STUDY_OWNER is allowed to update)
+     *
+     * @param populationId      ID of the population to be updated.
      * @param updatedPopulation Updated population model.
-     * @param studyId ID of the study.
-     * @param principal KeycloakPrincipal object that holds access token.
+     * @param studyId           ID of the study.
+     * @param principal         Jwt principal containing user info.
      * @return ResponseEntity with the updated population data.
      */
     @PutMapping("/{populationId}")
     public ResponseEntity<?> updatePopulation(@PathVariable Long populationId,
                                               @RequestBody Population updatedPopulation,
                                               @RequestParam Long studyId,
                                               @AuthenticationPrincipal Jwt principal) {
-        // Allowed roles for this endpoint
         List<Role> lesserAllowedRoles = List.of(Role.STUDY_OWNER);
-        // Check user authorization for the given study
         if (!this.roleCheckerService.isUserAuthorizedForStudy(studyId, principal, lesserAllowedRoles)) {
             return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
         }
 
         try {
-            Optional<Population> savedPopulation = this.populationService.updatePopulation(populationId, updatedPopulation);
-            return savedPopulation.map(ResponseEntity::ok).orElseGet(() -> ResponseEntity.notFound().build());
+            Optional<Population> savedPopulationOpt =
+                    this.populationService.updatePopulation(populationId, updatedPopulation);
+
+            if (savedPopulationOpt.isPresent()) {
+                Population savedPopulation = savedPopulationOpt.get();
+                String recordId = savedPopulation.getPopulationId().toString();
+                String description = "Update of Population " + recordId;
```

Comment:
I think we can create an enumeration for the descriptions and the pass this enum value to createAuditLog function along with a list of parameters (recordId in this case), wdyt ?